casper/contracts/simple_casper.v.py

```diff
@@ -432,12 +451,35 @@ def slash(vote_msg_1: bytes <= 1024, vote_msg_2: bytes <= 1024):
         # NO SURROUND VOTE
         slashing_condition_detected = True
     assert slashing_condition_detected
-    # Delete the offending validator, and give a 4% "finder's fee"
-    validator_deposit = self.get_deposit_size(validator_index_1)
-    slashing_bounty = validator_deposit / 25
-    self.total_destroyed += validator_deposit * 24 / 25
-    self.delete_validator(validator_index_1)
-    send(msg.sender, slashing_bounty)
+    # Check that the votes were valid
+    dynasty_of_vote_1 = self.dynasty_in_epoch[target_epoch_1]
+    dynasty_of_vote_2 = self.dynasty_in_epoch[target_epoch_2]
+    ok = (self.validators[validator_index_1].start_dynasty <= dynasty_of_vote_1 and \
+        dynasty_of_vote_1 <= self.validators[validator_index_1].end_dynasty) and \
+        (self.validators[validator_index_1].start_dynasty <= dynasty_of_vote_2 and \
+        dynasty_of_vote_1 <= self.validators[validator_index_2].end_dynasty)
+    assert ok
+    td1 = self.total_deposits[dynasty_of_vote_1]
+    td2 = self.total_deposits[dynasty_of_vote_2]
+    effective = False
+    if not self.validator_slashed_at_link[validator_index_1][source_epoch_1][target_epoch_1]:
+        self.validator_slashed_at_link[validator_index_1][source_epoch_1][target_epoch_1] = True
+        self.slashings_at_link[source_epoch_1][target_epoch_1] += self.validators[validator_index_1].deposit / td1
```

Comment:
Given dense code block, worth commenting that we are incrementing the % of slashings at the link by deposit / td.